core/src/main/java/org/elasticsearch/common/geo/GeoUtils.java

```diff
@@ -419,6 +428,8 @@ public static GeoPoint parseGeoPoint(XContentParser parser, GeoPoint point) thro
                 } else {
                     return point.resetFromGeoHash(geohash);
                 }
+            } else if (numberFormatException != null) {
+                throw new ElasticsearchParseException("[{}] and [{}] must be valid double values", LATITUDE, LONGITUDE);
```

Comment:
This swallows the numberFormatException but should instead be wrapped by the `ElasticsearchParseException`.